glue_job.tf

```diff
@@ -0,0 +1,233 @@
+resource "aws_glue_catalog_database" "catalog_db" {
+  name = "controlshift_${var.controlshift_environment}"
+}
+
+locals {
+  signatures_s3_path = "s3://agra-data-exports-${var.controlshift_environment}/${var.controlshift_organization_slug}/full/signatures"
+}
+
+resource "aws_glue_crawler" "signatures_crawler" {
+  database_name = aws_glue_catalog_database.catalog_db.name
+  name = "${var.controlshift_environment}_full_signatures"
+  role = aws_iam_role.glue_service_role.arn
+
+  s3_target {
+    path = local.signatures_s3_path
+  }
+}
+
+resource "aws_s3_bucket" "glue_resources" {
+  bucket = var.glue_scripts_bucket_name
+  region = var.aws_region
+
+  acl = "private"
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+}
+
+data "template_file" "signatures_script" {
+  template = file("${path.module}/templates/signatures_job.py.tpl")
+  vars = {
+    catalog_database_name = aws_glue_catalog_database.catalog_db.name
+    redshift_database_name = var.redshift_database_name
+    redshift_connection_name = aws_glue_connection.redshift_connection.name
+  }
+}
+
+resource "aws_s3_bucket_object" "signatures_script" {
+  bucket = aws_s3_bucket.glue_resources.id
+  key = "${var.controlshift_environment}/signatures_job.py"
+  acl = "private"
+
+  content = data.template_file.signatures_script.rendered
+}
+
+resource "aws_iam_role" "glue_service_role" {
+  name = "AWSGlueServiceRole"
+  description = "Used by the AWS Glue jobs to insert data into redshift"
+  assume_role_policy = data.aws_iam_policy_document.glue_assume_role.json
+}
+
+data "aws_iam_policy_document" "glue_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type = "Service"
+      identifiers = ["glue.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "glue_resources" {
+  role       = aws_iam_role.glue_service_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+}
+
+resource "aws_iam_role_policy_attachment" "redshift_full_access" {
+  role       = aws_iam_role.glue_service_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonRedshiftFullAccess"
```

Comment:
Not required for merge, but less permissions seems nice if possible.

Reply:
Yeah, will very likely work with less permissions. Will check docs to see if I can find the exact set of permissions required by Glue job.

Reply:
Actually, I think this is not even required. The glue job interaction with Redshift is done over the JDBC connection, and IAM doesn't do anything there.